src/hotspot/share/oops/method.cpp

```diff
@@ -2256,7 +2256,7 @@ bool Method::is_valid_method(const Method* m) {
 
 #ifndef PRODUCT
 void Method::print_jmethod_ids(const ClassLoaderData* loader_data, outputStream* out) {
-  out->print(" jni_method_id count = %d", loader_data->jmethod_ids()->count_methods());
+  out->print("%d", loader_data->jmethod_ids()->count_methods());
 }
```

Comment:
This method is pointless if it only prints the count to the stream without any descriptive text. Just delete it and print the method count directly in the caller.

Reply:
The type of jmethod_ids  is JNIMethodBlock, which is declared in `method.cpp`, we have to access its internal fields via print_jmethod_ids that declared in `method.hpp`.  But if we only want to show jmethod_ids's address, then we can remove Method::print_jmethod_ids and print it directly. I'm not sure if others think jmethod_ids._count is important for their debugging, so I keep unchanged :)

Reply:
Sorry I don't see your point. I am suggesting that this:

out->print   (" - jmethod count       ");
Method::print_jmethod_ids(this, out);
out->print_cr("");

be replaced by:

out->print_cr(" - jmethod count       %d", this->jmethod_ids()->count_methods());

Reply:
Hi David,

I means, `count_methods()` is inaccessible from classLoaderDataGraph.cpp because `this->jmethod_ids()` is type of JNIMethodBlock, it was defined in **method.cpp**, i.e.:
```cpp
// method.cpp
class JNIMethodBlock {
...
public:
int count_methods(){...}
}
```
In order to access count_methods, we have to add an API in method.hpp and method.cpp:
```cpp
// method.cpp
class JNIMethodBlock {
...
int count_methods(){...}
}

void Method::print_jmethod_ids(JNIMethodBlock* block,...){
  block->count_methods();
  ....
}
// method.hpp
void Method::print_jmethod_ids(JNIMethodBlock* block,...);
```
Thanks~
Yang

Reply:
Ah I see. In that case I'd suggest adding an API to get the count. If external code knows enough about jmethod_ids that it knows to print the count, then the count is worth exposing IMO. Otherwise restore the actual "jmethod count" text to the print function so it's a stand-alone print function again.
Aside: print_jmethod_ids() is badly named given it doesn't print any id's.